spec/features/demo_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+require 'database_dump_loader'
+
+RSpec.describe 'demo', type: :feature do
+
+  Warden.test_mode!
+
+  before do
+    # Load a snapshot of production data into the test database
+    DatabaseDumpLoader.new(File.read(File.expand_path('demo-data.sql', File.dirname(__FILE__)))).load
+  end
+
+  it "should be possible to demo the application" do
+    visit '/'
+
+    expect(page.current_path).to eql("/")
+    expect(page.title).to eql("GovAuBeta")
+    expect(page).to have_css('h2', :text => 'Information and services')
+    expect(page).to have_css('h2', :text => 'About Australia')
+    visit(page.find('.next-holiday a')[:href])
+
+    expect(page.current_path).to eql("/times-and-dates/australian-public-holidays")
+    expect(page).to have_css('h1', :text => 'Australian Public holidays')
+    visit(page.find('a', :text => 'School holidays and term dates')[:href])
+
+    expect(page.current_path).to eql("/times-and-dates/school-holidays-and-term-dates")
+    expect(page).to have_css('h1', :text => 'School holidays and term dates')
+    visit(page.find('nav.breadcrumbs--inverted a', :text => 'Home')[:href])
+
+    expect(page.current_path).to eql("/")
+    expect(page).to have_css('section.about-government h2', :text => 'About the Australian Government')
+    visit(page.find('a', :text => /18 Departments/)[:href])
+
+    expect(page.current_path).to eql("/departments")
+    visit(page.find('article.content-listing li a', :text => 'Department of Communications and the Arts')[:href])
+
+    expect(page.current_path).to eql('/department-of-communications-and-the-arts')
+    visit(page.find('aside li a', :text => 'Our role')[:href])
+
+    expect(page.current_path).to eql('/department-of-communications-and-the-arts/our-role')
+    visit(page.find('aside li a', :text => 'Careers')[:href])
+
+    expect(page.current_path).to eql('/department-of-communications-and-the-arts/careers')
+    visit(page.find('aside li a', :text => 'Our role')[:href])
```

Comment:
I think these precise selectors make the test too closely tied to the implementation = too fragile. Should only go to that level of detail if there's actual ambiguity on the page (e.g. more than one 'Our role' link).